src/validators/string.rs

```diff
@@ -144,6 +147,16 @@ impl StrConstrainedValidator {
         let to_upper: bool =
             schema_or_config(schema, config, intern!(py, "to_upper"), intern!(py, "str_to_upper"))?.unwrap_or(false);
 
+        let name = match config {
+            Some(_config) => Self::EXPECTED_TYPE.into(),
+            _ => format!(
+                "{}[max_length={} min_length={}]",
+                Self::EXPECTED_TYPE,
+                max_length.unwrap_or_default(),
+                min_length.unwrap_or_default()
+            ),
```

Comment:
No, we need to combine all struct constraints, and not include `max_length` and `min_length` if they're `None`, also min length should come before max length.

`strip_whitespace`, `to_lower` and `to_upper` should only be included if they're true.